projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/cartesian-widget-renderer.component.ts

```diff
@@ -114,6 +156,25 @@ export class CartesianWidgetRendererComponent<TSeriesInterval> extends Interacti
 
     return match || 'AUTO';
   }
+
+  private showContextMenu(): void {
+    this.popover = this.popoverService.drawPopover({
+      componentOrTemplate: this.contextMenuTemplate,
```

Comment:
You can pass the component here too. So, in the interaction handler, just pass a new component that has above  content